Dynatrace/dynatrace-operator
# Bugfix/always pass proxy to oa even when empty

## Description

When a proxy has been configured in a DynaKube previously and that proxy settings has been reomoved, OneAgents would not take into account that change, as they would still use their cached settings. We need to not only remove the proxy setting we pass to the OneAgent but explicityl set it as "empty".

## How can this be tested?
- deploy the operator
- setup a proxy in your cluster
- apply a DynaKube that uses that proxy
- deploy a test application to also include code modules in your test
- check if OneAgents and code modules are connecting to your tenant
  * Note: OneAgents can'r resolve a service address, use proxy IP
- check if code modules can connect (use agent registry in debug UI)
- remove proxy and see if agents loose the connection
- remove proxy from Dynakube config and re-apply
- restart app pods
- wait a bit and see if OneAgents and code modules are able to connect again

## Checklist
